File: app/src/main/java/com/example/dothingandroid/Connection.java
//package com.example.dothingandroid;
//
//import java.io.BufferedReader;
//import java.io.IOException;
//import java.io.InputStreamReader;
//import java.io.PrintWriter;
//import java.net.Socket;
//import java.util.ArrayList;
//import java.util.Scanner;
//import java.util.List;
//
//class Connection{
//    String ip;
//    Integer port;
//    Integer buffer;
//    PrintWriter writer;
//    BufferedReader reader;
//    Socket s;
//    Scanner scanner;
//    public Connection(String ip_in, Integer port_in, Integer buffer_in) throws IOException{
//        ip = ip_in;
//        port = port_in;
//        buffer = buffer_in;
//    }
//    public void connect() throws IOException {
//        this.s = new Socket(this.ip, this.port);
//        this.writer = new PrintWriter(this.s.getOutputStream(), true);
//        this.reader = new BufferedReader(new InputStreamReader(this.s.getInputStream()));
//        this.scanner = new Scanner(reader);
//    }
//    public void send(String message){
//        this.writer.println(message);
//    }
//    public String recv(){
//        if (this.scanner.hasNext()) {
//            return this.scanner.next();
//        }
//        else{
//            return "";
//        }
//    }
//    public String WaitUntilRecv(){
//        while(true){
//            String message = recv();
//            if (!message.equals("")){
//                return message;
//            }
//        }
//    }
//    public List<String> RecvList(String contcode, String endcode){
//        List<String> out = new ArrayList<String>();
//        while (true){
//            String data = this.recv();
//            this.send(contcode);
//            if (data.equals(endcode)){
//                break;
//            }
//            else if (!data.equals("")){
//                out.add(data);
//            }
//        }
//        return out;
//    }
//    public void dc() throws IOException {
//        this.s.close();
//    }
//}

File: app/src/main/java/com/example/dothingandroid/Connection.kt
package com.example.dothingandroid

import android.util.Log
import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.PrintWriter
import java.net.Socket
import java.util.*


class Connection(in_address: String, in_port: Int) {

    val address = in_address
    val port = in_port


    private var s: Socket = Socket(address, port)
    private var reader: BufferedReader = BufferedReader(InputStreamReader(s.getInputStream()))
    private var writer: PrintWriter = PrintWriter(s.getOutputStream(), true)
    private var scanner: Scanner = Scanner(reader)

    fun send(message: String, spacesafe: Boolean = false) {
        if (spacesafe) {
            writer.println(message.replace(" ", "_"))
        } else {
            writer.println(message)
        }
    }

    fun recv(): String {
        var output: String = ""
        if (scanner.hasNext()) {
            output = scanner.nextLine()
            Log.d("DEBUG", output)
        } else {
            output = ""
        }
        return output
    }

    fun WaitUntilRecv(): String {
        while (true) {
            val message = recv()
            if (message != "") {
                return message
            }
        }
    }

    fun RecvList(contcode: String, endcode: String): MutableList<String> {
        val out: MutableList<String> = ArrayList()
        while (true) {
            val data = recv()
            send(contcode)
            if (data == endcode) {
                break
            } else if (data != "") {
                out.add(data)
            }
        }
        return out
    }

    fun SendList(endcode: String, list: List<String>) {
        for (item in list) {
            send(item, spacesafe = true)
            WaitUntilRecv()
        }
        send(endcode)
    }

    fun dc() {
        s.close()
    }

}